db/hints/manager.cc

```diff
@@ -352,10 +371,15 @@ bool manager::store_hint(endpoint_id ep, schema_ptr s, lw_shared_ptr<const froze
 bool manager::too_many_in_flight_hints_for(endpoint_id ep) const noexcept {
     // There is no need to check the DC here because if there is an in-flight hint for this
     // endpoint, then this means that its DC has already been checked and found to be ok.
-    return _stats.size_of_hints_in_progress > MAX_SIZE_OF_HINTS_IN_PROGRESS
+    if (!(_stats.size_of_hints_in_progress > MAX_SIZE_OF_HINTS_IN_PROGRESS
```

Comment:
Yes, that was accidental.
Feel free to write a test for this function.

Reply:
Just to clarify: the PR inverts the condition because now the `if` enters a branch which returns `false`. Benny's commit didn't invert the whole `if`, only one of the conjuncts, i.e. there is a missing `!` before `_proxy.local_db().get_token_metadata().get_topology().is_me(ep)`.

I created an issue: https://github.com/scylladb/scylladb/issues/17636

As for writing a test... with error injection blocking storing hints to disk it should be possible to write a test indeed.

Reply:
I also created an issue but forgot to post it here https://github.com/scylladb/scylladb/issues/17129. Will close as duplicate